app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -64,6 +70,17 @@ public void onClick(View view) {
         return view;
     }
 
+    @Override public void onStart() {
+        super.onStart();
+        _habitItemAdapter.startListening();
+    }
+
+    @Override public void onStop()
+    {
+        super.onStop();
+        _habitItemAdapter.stopListening();
+    }
```

Comment:
These two methods prevent the app from **always** fetching new habits. We only want to fetch on start, and stop fetching when the app stops.